src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs

```diff
@@ -570,5 +576,85 @@ public uint GetTokens(uint dwMaxTokens, IVsSearchToken[] rgpSearchTokens)
                 return 0;
             }
         }
+
+        private class PackageInstallerServicesProxy : IPackageInstallerServicesProxy
```

Comment:
Ah.  I do like that.

             -- Cyrus

From: Kevin Pilch-Bisson [mailto:notifications@github.com]
Sent: Thursday, May 19, 2016 6:02 PM
To: dotnet/roslyn <roslyn@noreply.github.com>
Cc: Cyrus Najmabadi <cyrusn@microsoft.com>; Author <author@noreply.github.com>
Subject: Re: [dotnet/roslyn] Delay loading Nuget dlls until we actually need them. (#11440)


In src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs<https://github.com/dotnet/roslyn/pull/11440#discussion_r63977870>:

> @@ -570,5 +576,85 @@ public uint GetTokens(uint dwMaxTokens, IVsSearchToken[] rgpSearchTokens)

>                  return 0;

>              }

>          }

> +

> +        private class PackageInstallerServicesProxy : IPackageInstallerServicesProxy

Well, instead of creating proxy interfaces 1:1 with the existing NuGet interfaces, you could instead make a single interface that has the API you want, and have a single class that implements that, by wrapping the 4 interfaces from NuGet.

(Aka - the adapter pattern<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fen.wikipedia.org%2fwiki%2fAdapter_pattern&data=01%7c01%7ccyrusn%40microsoft.com%7ca6779c0dbe594af9876b08d3804a6390%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=l0HBDvwZNVobPCml9ddGr4DMypHggZFBqFLXNRZynxc%3d>)

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/dotnet/roslyn/pull/11440/files/35449fa6d56bfeee6ba06f06d925fed6e364fe59#r63977870>